File: app/src/main/java/com/loopeer/codereader/ui/loader/RecyclerLoader.java
package com.loopeer.codereader.ui.loader;

import android.widget.ViewAnimator;

public class RecyclerLoader implements ILoadHelper {

    ViewAnimator mViewAnimator;

    public RecyclerLoader(ViewAnimator continer) {
        mViewAnimator = continer;
    }

    @Override
    public void showProgress() {
        mViewAnimator.setDisplayedChild(2);
    }

    @Override
    public void showContent() {
        mViewAnimator.setDisplayedChild(0);
    }

    @Override
    public void showEmpty() {
        mViewAnimator.setDisplayedChild(1);
    }
}

File: code-reader-kt/src/main/java/com/loopeer/codereaderkt/ui/loader/RecyclerLoader.kt
package com.loopeer.codereaderkt.ui.loader

import android.widget.ViewAnimator


class RecyclerLoader(internal var mViewAnimator: ViewAnimator) : ILoadHelper {

    override fun showProgress() {
        mViewAnimator.displayedChild = 2
    }

    override fun showContent() {
        mViewAnimator.displayedChild = 0
    }

    override fun showEmpty(message: String) {
        mViewAnimator.displayedChild = 1
    }
}